Dockerfile

```diff
@@ -1,8 +1,35 @@
 # Docker Container that runs the Nebra Diagnostics Tool
 
-FROM balenalib/raspberry-pi-debian-python:buster-run-20211014
+####################################################################################################
+################################## Stage: builder ##################################################
 
-WORKDIR /opt/
+FROM balenalib/raspberry-pi-debian-python:buster-run-20211014 as builder
+
+ENV PYTHON_DEPENDENCIES_DIR=/opt/python-dependencies
+
+RUN mkdir /tmp/build
+COPY ./ /tmp/build
+WORKDIR /tmp/build
+
+RUN \
+    install_packages \
+            build-essential \
+            libdbus-glib-1-dev && \
+    pip3 install --no-cache-dir --target="$PYTHON_DEPENDENCIES_DIR" .
```

Comment:
Nitpicking here, but I'd probably run this on a separate `RUN` line just to help with debugging if for whatever reason it would fail building.

Reply:
It would help debugging.

Reply:
> Approved by me, but I would ideally like to see some unit tests here. Don't merge this before @marvinmarnold approves it.

Yeah, I am writing the unit test and a parallel logic in `DiagnosticsReport`. These pushes are just for `In-progress`.

Reply:
I see. Please update the PR to 'Draft' mode if that is the case.